bindings/web5_c/src/c.rs

```diff
@@ -10,6 +10,52 @@ pub extern "C" fn free_string(s: *mut c_char) {
     }
 }
 
+use tokio::runtime::Runtime;
+use web5::errors::Result;
+use web5::errors::Web5Error;
+
+pub fn get_rt() -> Result<Runtime> {
+    let rt = Runtime::new()
+        .map_err(|e| Web5Error::Unknown(format!("unable to instantiate tokio runtime {}", e)))?;
+    Ok(rt)
+}
+
+use serde_json;
+use std::ptr;
+use web5::dids::methods::did_dht;
+
+#[no_mangle]
+pub extern "C" fn did_dht_resolve(uri: *const c_char, gateway_url: *const c_char) -> *mut c_char {
+    let uri = match unsafe { CStr::from_ptr(uri).to_str() } {
+        Ok(s) => s,
+        Err(_) => return ptr::null_mut(),
+    };
+
+    let gateway_url = if gateway_url.is_null() {
+        None
+    } else {
+        match unsafe { CStr::from_ptr(gateway_url).to_str() } {
+            Ok(s) => Some(s.to_string()),
+            Err(_) => return ptr::null_mut(),
```

Comment:
at some point we should introduce better error handling over the C FFI, but it's not necessary at this moment

Reply:
Created https://github.com/TBD54566975/web5-rs/issues/392 to track.